### PR TITLE
remove dependency matrix from `asv.ci.conf.json`

### PR DESCRIPTION
**Pull request by zacharyburnett**
### Description

tests changes to benchmarking environment from https://github.com/astropy/astropy-benchmarks/pull/113

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: .github/workflows/ci_benchmark.yml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
           set -x
           python -m pip install asv virtualenv packaging
 
-          git clone -b main https://github.com/astropy/astropy-benchmarks.git --single-branch
+          git clone -b main https://github.com/zacharyburnett/astropy-benchmarks.git --single-branch -b fix/asv_conf
```
Comment (pllim) [inline .github/workflows/ci_benchmark.yml:76]: No need to call `-b` twice?
```suggestion
          git clone -b fix/asv_conf https://github.com/zacharyburnett/astropy-benchmarks.git --single-branch
```

**Comment by pllim** [comment]
Hmm wasn't expecting this:

```
| Change   | Before [c638d70a]    | After [0c5ca84c]    |   Ratio | Benchmark (Parameter)                |
|----------|----------------------|---------------------|---------|--------------------------------------|
| +        | 38.1±0.04μs          | 77.6±40μs           |    2.04 | units.time_quantity_creation         |
| +        | 92.5±0.3μs           | 152±60μs            |    1.65 | units.time_quantity_init_array       |
| +        | 157±0.6μs            | 222±60μs            |    1.41 | units.time_quantity_array_conversion |

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
Error: PERFORMANCE DECREASED.
```

Maybe https://github.com/astropy/astropy/blob/main/asv.ci.conf.json needs update too to go with your PR?

**Comment by pllim** [comment]
Interesting... 🤔 😸 

```
| Change   | Before [c638d70a]    | After [0a37090e]    |   Ratio | Benchmark (Parameter)                                         |
|----------|----------------------|---------------------|---------|---------------------------------------------------------------|
| -        | 306±9μs              | 228±0.4μs           |    0.75 | units.TimeQuantityOpLargeArraySameUnit.time_quantity_np_equal |
| -        | 462±10μs             | 339±10μs            |    0.73 | units.TimeQuantityOpLargeArraySameUnit.time_quantity_np_add   |

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE INCREASED.
```

**Comment by pllim** [comment]
How do I know just by inspecting the log (without the UI) what versions of what were used in the benchmarks here? I cannot figure it out looking at https://github.com/astropy/astropy/actions/runs/8722480045/job/23928443193?pr=16296

**Comment by zacharyburnett** [comment]
> How do I know just by inspecting the log (without the UI) what versions of what were used in the benchmarks here? I cannot figure it out looking at https://github.com/astropy/astropy/actions/runs/8722480045/job/23928443193?pr=16296

I'm not sure that is mentioned in the log; you would have to check the results files

**Comment by pllim** [comment]
Hmm I see artifact at https://github.com/astropy/astropy/actions/runs/8722480045/artifacts/1422151899 from https://github.com/astropy/astropy/actions/runs/8722480045/job/23928443193?pr=16296 but it does not mention dependency versions.

**Comment by zacharyburnett** [comment]
> Hmm I see artifact at https://github.com/astropy/astropy/actions/runs/8722480045/artifacts/1422151899 from https://github.com/astropy/astropy/actions/runs/8722480045/job/23928443193?pr=16296 but it does not mention dependency versions.

ah I see now, the results artifact doesn't actually include the results; I reran with it uploading

**Comment by zacharyburnett** [comment]
the results will only show dependency versions if specified in the `matrix.req`; we can include `matplotlib` and `scipy` to `matrix.req` if it is necessary to freeze versions they are run against

**Comment by pllim** [comment]
Thanks! But I still cannot find it. Maybe searching for "numpy" isn't enough? Are you able to?

https://github.com/astropy/astropy/actions/runs/8725413079/artifacts/1422883208

**Comment by zacharyburnett** [comment]
here: https://github.com/astropy/astropy/actions/runs/8738886254

you can see the explicit `numpy` and `matplotlib` versions in these results because they're defined in the `matrix.reqs`

**Comment by pllim** [comment]
I was thinking more like the pytest header we have or at least a `pip freeze` somewhere in the log. Is that possible?

Anyway, I think that is getting out of topic. The conclusion from this PR is that your https://github.com/astropy/astropy-benchmarks/pull/113 is acceptable but only if we also update the JSON here ASAP afterwards.

Thanks!

File: .github/workflows/ci_benchmark.yml [hunk 2]
```diff
@@ -73,7 +73,7 @@ jobs:
           set -x
           python -m pip install asv virtualenv packaging
 
-          git clone -b main https://github.com/astropy/astropy-benchmarks.git --single-branch
+          git clone -b fix/asv_conf https://github.com/zacharyburnett/astropy-benchmarks.git --single-branch
```
Comment (pllim) [inline .github/workflows/ci_benchmark.yml:76]: If you undo this line, I think we can then bring it out of draft and get it in. Thanks!

**Comment by pllim** [comment]
RTD failure is unrelated, so I am gonna squash merge. Thanks!

**Review by pllim — APPROVED** [review]

**Pull request merged**